### ISSUE TITLE
Incompatible with latest version of httpx (TypeError error when uploading image: Expected bytes or bytes-like object got: <class 'str'>)

### ISSUE THREAD

**Issue by dudududodododedede** [jackm/kijiji-manager#16]
There seems to be a regression with httpx 0.19.0

I see that requirements.txt has "httpx==0.16.1" but when I did "pip install ." (step 1.3 of the readme) it installed  0.19.0 which was just release 18 days ago

This is the error I get when trying to post an ad using Kijiji Manager

Internal Server Error

console:
```127.0.0.1 - - [06/Sep/2021 13:01:36] "POST /post HTTP/1.1" 200 -
127.0.0.1 - - [06/Sep/2021 13:01:36] "GET /static/style2.css HTTP/1.1" 304 -
127.0.0.1 - - [06/Sep/2021 13:01:36] "GET /loc?location1=9004 HTTP/1.1" 200 -
127.0.0.1 - - [06/Sep/2021 13:01:37] "GET /loc?location1=9004&location2=1700006 HTTP/1.1" 200 -
127.0.0.1 - - [06/Sep/2021 13:02:12] "GET /loc?location1=9004&location2=1700272 HTTP/1.1" 200 -
[2021-09-06 13:02:58,474] ERROR in app: Exception on /post [POST]
Traceback (most recent call last):
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/kijiji_manager/views/ad.py", line 234, in post
    'pic:pictures': create_picture_payload(form.data),
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/kijiji_manager/views/ad.py", line 392, in create_picture_payload
    link = kijiji_api.upload_image(value)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/kijiji_manager/kijijiapi.py", line 270, in upload_image
    r = self.session.post(api_endpoint, headers=headers, files=files)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_client.py", line 1109, in post
    return self.request(
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_client.py", line 776, in request
    request = self.build_request(
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_client.py", line 346, in build_request
    return Request(
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_models.py", line 1097, in __init__
    headers, stream = encode_request(content, data, files, json)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_content.py", line 179, in encode_request
    return encode_multipart_data(data or {}, files, boundary)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_content.py", line 124, in encode_multipart_data
    multipart = MultipartStream(data=data, files=files, boundary=boundary)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_multipart.py", line 161, in __init__
    self.fields = list(self._iter_fields(data, files))
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_multipart.py", line 175, in _iter_fields
    yield FileField(name=name, value=value)
  File "/Users/user/Downloads/kijiji-manager/venv/lib/python3.9/site-packages/httpx/_multipart.py", line 86, in __init__
    raise TypeError(f"Expected bytes or bytes-like object got: {type(fileobj)}")
TypeError: Expected bytes or bytes-like object got: <class 'str'>
127.0.0.1 - - [06/Sep/2021 13:02:58] "POST /post HTTP/1.1" 500 -
```

I believe the error comes from this commit: https://github.com/encode/httpx/blame/c24bbb85a6efd83beb2ed6aab37913e4e0d44418/httpx/_multipart.py#L86

**Comment by dudududodododedede** [comment]
This fixed it for me:
```
pip uninstall httpx
pip install -Iv httpx==0.18.2
```

**Comment by jackm** [comment]
Yes it does appear that starting with [httpx version 0.19.0](https://github.com/encode/httpx/blob/master/CHANGELOG.md#0190-19th-august-2021) a breaking [change](https://github.com/encode/httpx/pull/1736) was made that forces file upload content to be `bytes` type and won't accept `str` type anymore.

Should be an easy fix, but I'd like to do more testing first.

**Comment by jackm** [comment]
Fixed in 9ab128e
